src/main/java/bleuauction/bleuauction_be/server/notice/service/NoticeService.java

```diff
@@ -1,27 +1,53 @@
 package bleuauction.bleuauction_be.server.notice.service;
 
+import bleuauction.bleuauction_be.server.attach.entity.Attach;
+import bleuauction.bleuauction_be.server.attach.service.AttachService;
+import bleuauction.bleuauction_be.server.member.entity.Member;
+import bleuauction.bleuauction_be.server.member.entity.MemberCategory;
+import bleuauction.bleuauction_be.server.ncp.NcpObjectStorageService;
 import bleuauction.bleuauction_be.server.notice.entity.Notice;
 import bleuauction.bleuauction_be.server.notice.entity.NoticeStatus;
 import bleuauction.bleuauction_be.server.notice.repository.NoticeRepository;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
 
-import java.sql.Timestamp;
-import java.time.LocalDateTime;
+import java.util.ArrayList;
 import java.util.List;
 
+import static bleuauction.bleuauction_be.server.member.entity.MemberCategory.A;
+
 @Service
 @Transactional(readOnly = true)
 @RequiredArgsConstructor
 public class NoticeService {
 
   private final NoticeRepository noticeRepository;
+  private final NcpObjectStorageService ncpObjectStorageService;
+  private final AttachService attachService;
 
   @Transactional
-  public Long enroll(Notice notice) {
-    noticeRepository.save(notice);
+  public Long enroll(Notice notice, List<MultipartFile> multipartFiles, Member member) {
+
+    if(member.getMemberCategory() == MemberCategory.A) {
+      notice.setMemberNo(member);
+      noticeRepository.save(notice);
+
+      if (multipartFiles != null && multipartFiles.size() > 0) {
+        ArrayList<Attach> attaches = new ArrayList<>();
+        for (MultipartFile multipartFile : multipartFiles) {
+          if (multipartFile.getSize() > 0) {
+            Attach attach = ncpObjectStorageService.uploadFile(new Attach(),
+                    "bleuauction-bucket", "notice/", multipartFile);
+            notice.addNoticeAttach(attach);
+          }
+        }
+      }
+
+    }
     return notice.getNoticeNo();
+
   }
```

Comment:
동일한 기능인데 전 이렇게 할 것 같아요.

if문을 계속 타고 들어가면 가독성이 떨어지다 보니 최대한 제거하는 방향으로 작성을 하려고 전 지향합니다 ㅎㅎ :)

번외로 스트림 익혀보는것도 권장합니다 ㅋㅋ
```suggestion
    public Long enroll(Notice notice, List<MultipartFile> multipartFiles, Member member) {
        if (!MemberCategory.A.equals(member.getMemberCategory())) {
            // 관리자가 아니기 떄문에 예외처리 로직
        }

        notice.setMemberNo(member);
        if (multipartFiles != null && !multipartFiles.isEmpty()) {
            multipartFiles.stream()
                    .filter(file -> file.getSize() > 0)
                    .forEach(multipartFile ->
                            notice.addNoticeAttach(ncpObjectStorageService.uploadFile(new Attach(),
                                    "bleuauction-bucket", "notice/", multipartFile))
                    );
        }
        return noticeRepository.save(notice).getNoticeNo();
    }
```